### PR TITLE
SurveyJS Multi Language Support

### PR DESCRIPTION
**Pull request by edwardstamps**

### CODE REVIEW

File: mobile/src/SurveyTable.js [hunk 1]
```diff
@@ -131,7 +131,8 @@ export default class SurveyTable extends Component {
 
   returnName = item => {
     const parsedData = JSON.parse(item.info)
-    return parsedData.title
+    const title = typeof parsedData.title === 'object' ? parsedData.title.default : parsedData.title
```
Comment (AdamLiechty) [inline mobile/src/SurveyTable.js]: extract this logic into a reusable function

File: web/admin/src/SurveyResults.js [hunk 2]
```diff
@@ -179,8 +179,9 @@ class SurveyResults extends Component {
       })
       origQuestions.forEach(question => {
         const name = question.name.replace(/\.$/, '')
+        const title = typeof question.title === 'object' ? question.title.default : question.title
         headers.push({
-          label: question.title ? question.title.trim() : question.name.trim(),
```
Comment (AdamLiechty) [inline web/admin/src/SurveyResults.js]: looks like this loses the trimming of the title

File: mobile/src/home-view.js [hunk 3]
```diff
@@ -170,8 +170,12 @@ class HomeView extends PureComponent {
           return name === item.replace('-Comment', '')
         })
         if (question) {
-          const answer = JSON.stringify(origResults[item])
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: this should use your reusable function

File: web/admin/src/App.js [hunk 4]
```diff
@@ -237,6 +237,8 @@ class App extends PureComponent {
             publishedVersion.publishDate === a.publishDate
           : false
         const isPublishedHidden = isPublished && !publishedVersion.isViewable
+        const title =
+          typeof parsedData.title === 'object' ? parsedData.title.default : parsedData.title
```
Comment (AdamLiechty) [inline web/admin/src/App.js]: reusable function

File: web/admin/src/SurveyResults.js [hunk 5]
```diff
@@ -179,8 +179,9 @@ class SurveyResults extends Component {
       })
       origQuestions.forEach(question => {
         const name = question.name.replace(/\.$/, '')
+        const title = typeof question.title === 'object' ? question.title.default : question.title
```
Comment (AdamLiechty) [inline web/admin/src/SurveyResults.js]: reusable function

File: mobile/src/home-view.js [hunk 6]
```diff
@@ -170,8 +170,11 @@ class HomeView extends PureComponent {
...
-          let questionTitle = question.title ? question.title : question.name
+          const answer =
+            typeof origResults[item] === 'object'
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: This seems to be doing the same localization as for title, so let's reuse the same functionality

Comment (edwardstamps) [reply mobile/src/home-view.js]: Should I make a new function or edit the current one to be more universal as the title function  is checking a variable of the main object called title

Comment (AdamLiechty) [reply mobile/src/home-view.js]: Yeah, I'd say make the function something like:
```js
const getDefaultLocale = possiblyLocalized => typeof possiblyLocalized === 'object'
  ? possiblyLocalized.default
  : possiblyLocalized
```

File: mobile/src/home-view.js [hunk 7]
```diff
@@ -170,8 +170,8 @@ class HomeView extends PureComponent {
           return name === item.replace('-Comment', '')
         })
         if (question) {
-          const answer = JSON.stringify(origResults[item])
-          let questionTitle = question.title ? question.title : question.name
+          const answer = getDefaultLocale(origResults[item])
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: What was the JSON.stringify for?  We don't need it anymore?

Comment (edwardstamps) [reply mobile/src/home-view.js]: I believe this is an old fix for matrix questions acting as objects and causes a crash when questions (keys) had periods ending them

Comment (edwardstamps) [reply mobile/src/home-view.js]: Good catch

**Pull request merged**